pkg/minishift/config/config_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var testDir string
+
+func TestNewInstanceConfig(t *testing.T) {
+	setup(t)
+	defer teardown()
+
+	expectedFilePath := filepath.Join(testDir, "fake-machine.json")
+	cfg, _ := NewInstanceConfig(expectedFilePath)
+
+	if cfg.FilePath != expectedFilePath {
+		t.Errorf("Expected path '%s'. Received '%s'", expectedFilePath, cfg.FilePath)
+	}
+}
+
+func TestJsonFileExists(t *testing.T) {
+	setup(t)
+	defer teardown()
+
+	filePath := filepath.Join(testDir, "fake-machine.json")
+	expectedOcPath := filepath.Join(testDir, "fakeOc")
+	cfg := &InstanceConfig{
+		FilePath: filePath,
+		OcPath:   expectedOcPath,
+	}
+
+	jsonData, _ := json.MarshalIndent(cfg, "", "\t")
```

Comment:
@hferentschik , here I want to test `readInstanceConfig` being called from `NewInstanceConfig` when the json file exists.

For that, I am first creating json with appropriate data and calling `NewInstanceConfig` and checking this instance object's oc path which should be populated when creating json.

Hope you got the idea what I am trying to achieve here.